lib/bech32/src/Codec/Binary/Bech32/Internal.hs

```diff
@@ -90,7 +95,45 @@ import qualified Data.ByteString as BS
 import qualified Data.ByteString.Char8 as B8
 
 {-------------------------------------------------------------------------------
-                          Human Readable Parts
+                                 Data Part
+-------------------------------------------------------------------------------}
+
+-- | Represents the data part of a Bech32 string, as defined here:
+--   https://git.io/fj8FS
+newtype DataPart = DataPart
+    { getDataPart :: [Word5] }
+    deriving Eq
+
+instance Monoid DataPart where
+    mempty = DataPart []
+
+instance Semigroup DataPart where
+    (DataPart d0) <> (DataPart d1) = DataPart $ d0 <> d1
```

Comment:
Suggestion: use `DerivingStrategies` + `GeneralizedNewtypeDeriving`, or `DerivingVia` to get the job done by GHC :p

Reply:
Thanks for the suggestion. I've used `DerivingStrategies` to simplify this code.